lib/pos_leader_schedule.go

```diff
@@ -13,14 +14,13 @@ func (bav *UtxoView) GenerateLeaderSchedule() ([]*PKID, error) {
 		return nil, errors.Wrapf(err, "UtxoView.GenerateLeaderSchedule: error retrieving CurrentRandomSeedHash: ")
 	}
 
-	// Retrieve the SnapshotGlobalParamsEntry.LeaderScheduleMaxNumValidators.
-	snapshotGlobalParamsEntry, err := bav.GetSnapshotGlobalParamsEntry()
-	if err != nil {
-		return nil, errors.Wrapf(err, "UtxoView.GenerateLeaderSchedule: error retrieving SnapshotGlobalParamsEntry: ")
-	}
+	// Retrieve the LeaderScheduleMaxNumValidators from the current GlobalParams. We are safe to use the current
+	// global params because this generates a new leader schedule from the current validator entries, in preparation
+	// to snapshot the leader schedule.
+	currentGlobalParamsEntry := bav.GetCurrentGlobalParamsEntry()
```

Comment:
So I guess this was a bug before?

Reply:
Correct. This was a subtle bug. It would have resulted in an extra 2 epoch delay for leader schedule size changes to take effect. In that 2 epoch window, we would have had a mismatch between the leader schedule's actual size, and what the snapshotted global params say leader schedule size should be.

We would have hit this eventually when implementing the consensus.